### ISSUE TITLE
cuVS support for IDSelector APIs

### ISSUE THREAD

**Issue by cjnolet** [facebookresearch/faiss#4016]
This is a tracking issue to enable support for the IDSelector (e.g. pre-filtering) APIs through the cuVS indexes. We will contribute this feature, but want to have an issue to link.

**Comment by cjnolet** [comment]
> I don't know how feasible it is to shell out from GPU to call a virtual function for the general case. 

@mdouze, yes we plan to support the bare `IDSelector` types (like range and bitmap) initially. Longer term we'd like to find a good way to express more generalized filters on device (potentially with the use of JIT).

**Comment by mnorris11** [comment]
@cjnolet @tarang-jain is this issue being completed by #4858?
